modules/IntOGen.py

```diff
@@ -1,211 +1,190 @@
-                # loss_of_function_variant
-                .when(F.col('ROLE') == 'LoF', 'SO_0002054')
-                .otherwise(None)
+            logging.info(
+                f"Number of evidence without EFO mapping: {unmapped_evidence_count}"
             )
-        )
-
-        cohorts = (
-            self.spark
-            .read.csv(inputCohorts, sep=r'\t', header=True, inferSchema=True)
-            .select(
-                'COHORT',
-                'CANCER_TYPE_NAME',
-                'WEB_SHORT_COHORT_NAME',
-                'WEB_LONG_COHORT_NAME',
-                F.col('SAMPLES').alias('numberSamplesTested')
+            logging.info(
+                f"Number of diseases without EFO mapping: {unmapped_disease_count}"
             )
-        )
 
-        # Joining genes and cohorts data
-        self.dataframe = genes.join(
-            cohorts,
-            on='COHORT',
-            how='inner'
-        )
+        # Dropping cancer type acronym:
+        self.evidence = self.evidence.drop("Cancer_type_acronym")
 
-        # Mapping step
-        if not skipMapping:
-            try:
-                self.dataframe = self.cancer2EFO(diseaseMapping)
-                logging.info('Disease mappings have been imported.')
-            except Exception as e:
-                logging.error(f'An error occurred while importing disease mappings: \n{e}.')
-        else:
-            logging.info('Disease mapping has been skipped.')
-            self.dataframe = self.dataframe.withColumn(
-                'EFO_id',
-                F.lit(None)
-            )
+    def write_evidence(self: intogenEvidenceGenerator, evidence_file: str) -> None:
+        write_evidence_strings(self.evidence, evidence_file)
+        logging.info(
+            f"{self.evidence.count()} evidence strings saved into {evidence_file}. Exiting."
+        )
 
-        # Build evidence strings per row
-        logging.info('Generating evidence:')
-        evidences = (
-            self.dataframe.rdd
-            .map(intogenEvidenceGenerator.parseEvidenceString)
-            .collect()
-        )  # list of dictionaries
-
-        for evidence in evidences:
-            if skipMapping:
-                # Delete empty keys if mapping is skipped
-                del evidence['diseaseFromSourceMappedId']
-            if not evidence['mutatedSamples'][0]['functionalConsequenceId']:
-                # Dele empty key if functionalConsequenceId is unknown
-                del evidence['mutatedSamples'][0]['functionalConsequenceId']
-
-        return evidences
-
-    def cancer2EFO(self, diseaseMapping):
-
-        diseaseMappingsFile = (
-            self.spark
-            .read.csv(diseaseMapping, sep=r'\t', header=True)
-            .select('Cancer_type_acronym', 'EFO_id')
-            .withColumn('EFO_id', F.trim(F.col('EFO_id')))
+    def cancer2EFO(self: intogenEvidenceGenerator, diseaseMapping: str) -> DataFrame:
+        diseaseMappingsFile = self.spark.read.csv(
```

Comment:
Ammmm, rather not, this would complicate the logic as the `diseaseMapping` file is optional and the reading needs to be dealt with under an if branch, which does exists and only function is to call this function.